src/borg/archiver.py

```diff
@@ -107,6 +108,9 @@ def wrapper(self, args, **kwargs):
             with repository:
                 if manifest or cache:
                     kwargs['manifest'], kwargs['key'] = Manifest.load(repository)
+                    # do_recreate uses args.compression=None as in band signalling for "don't recompress"
+                    if getattr(args, 'compression', False):
+                        kwargs['key'].compressor = args.compression.compressor
```

Comment:
even with the comment, it is not quite clear how it works.

key.compressor is by default Compressor('none') if line 113 is not executed.

it is executed for a normal create run and then either lz4 or whatever was specified on the cmdline.

if args.compression is None (IBS) the line is not executed and we still have Compressor('none') there, which is "no/zero compression" but not "don't recompress".